### ISSUE TITLE
Real time object detection android app to deploy yolov5s.tflite model

### ISSUE THREAD

**Issue by HripsimeS** [ultralytics/yolov5#10431]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

@glenn-jocher Hello! 

Do you know any open source a real time object detection app in which is possible deploy **yolov5s.tflite** model ??
There are some apps that allows to take a picture and then do a detection. But unfortunately not so much available Android apps that you can do a object detection in real life with camara. Hope you have some more information, thanks in advance!

**Comment by AyushExel** [comment]
@HripsimeS Have you tried using the ultralytics app on android?

**Comment by HripsimeS** [comment]
@AyushExel thanks for your reply! No I did not try yet. Can you please some links on ultralytics app. Thanks in advance!

**Comment by AyushExel** [comment]
here you go https://ultralytics.com/app
You can also be redirected to the app via the QR code on our README  banner.
<img width="875" alt="Screenshot 2022-12-08 at 9 37 15 PM" src="https://user-images.githubusercontent.com/15766192/206498663-2597b66d-6a8c-4327-b15a-a9de53d952b2.png">

**Comment by HripsimeS** [comment]
@AyushExel thank you very much, looks very interesting. Is it also possible to deploy my yolov5s.tflite model trained on my dataset (not COCO) with different classes? If yes, can you please send some tips how to do it. Thanks a lot in advance :)

**Comment by glenn-jocher** [comment]
@HripsimeS yes you can train and deploy YOLOv5 models with custom datasets on the app at https://hub.ultralytics.com/

**Comment by glenn-jocher** [comment]
@AyushExel @pderrenger @HripsimeS good news 😃! PR #10446 updates our README to make our iOS and Android app more visible to users.

To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://bit.ly/yolov5-paperspace-notebook"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!

**Comment by HripsimeS** [comment]
@AyushExel @glenn-jocher Hello. I check in more details, in https://github.com/ultralytics/yolov5/blob/master/export.py script you are already adding a TFLite metadata which is great. But when I am trying to deploy the converted model to one of  MLKit apps, it does not work at all  https://github.com/googlesamples/mlkit/tree/master/android
Is it because the converted model doesn't fit the outputs requirements of their models (EfficientDet or MobileNet) ?

Please let me know how to fix the issue with **yolov5s.tflite** to be able to deploy in MLKit apps, thanks a lot in advance!

**Comment by AyushExel** [comment]
@HripsimeS try training your model through ultralytics hub. ([hub.ultrlaytics.com](https://hub.ultralytics.com/)) . This model will be accessible on your ultralytics app once trained. You can test it there and see if it works

**Comment by HripsimeS** [comment]
@AyushExel thank you very much for your reply. I trained my dataset on ultralytics hub and tested on some images, the prediction is great and training was very fast with 100 epoches (I assume it is not possible to change parameters).

Then I was checking how to integrate my trained model in ultralytics app, I could not figure out how to progress. Can you please tell me how I can deploy a model into app and check it with my phone. Thanks in advance!

![image](https://user-images.githubusercontent.com/42246765/206697022-711fced7-79fa-4d23-ae38-269f0a6b1e34.png)

**Comment by AyushExel** [comment]
@HripsimeS you should see that model available in your ultrlaytics app->projects section.

**Comment by HripsimeS** [comment]
@AyushExel it worked perfect, thanks!! I will close this issue 🥇

**Comment by glenn-jocher** [comment]
@HripsimeS you can update training settings in the 'Advanced Options' menu:

<img width="1407" alt="Screenshot 2022-12-09 at 15 40 20" src="https://user-images.githubusercontent.com/26833433/206812643-1a3b2b00-1e35-4ab2-be7d-846d0ccccdf9.png">

@kalenmike @sergiossm a happy HUB user :)

**Comment by kalenmike** [comment]
@glenn-jocher It seems that the HUB is not getting enough visibility, and the HUB Web is not providing enough visibility to the HUB app. We can and should address both of these, @sergiossm is working on wireframes for the latter.

**Comment by glenn-jocher** [comment]
@kalenmike yeah agree. I've added a big App banner in YOLOv5, but the README in general might benefit from a rethink. It's been built on previous versions so long that maybe it's worth a fresh look at how to do it.

**Comment by HripsimeS** [comment]
@glenn-jocher Hello. Would be possible to have an access or download the project files of ultralytics app (.gradle, .idea, app, build, gradle) ? Look forward to hearing from you!

**Comment by glenn-jocher** [comment]
@HripsimeS they're not currently open-source, but we're working on a way forward for mobile development. Could you describe your use case / end product / workflow in more detail to help us understand? Thanks!

**Comment by SaadAshraf12** [comment]
hiii, how do i access the code and all the packages of ultralytics android app?? In which language is the app created??

**Comment by glenn-jocher** [comment]
@SaadAshraf12 hello! The Ultralytics Android app is not open-source, so the code and packages are not publicly available. The app is part of our proprietary offerings to complement the YOLOv5 ecosystem. It's typically developed using Java or Kotlin for Android. If you have a specific use case or feature request, feel free to share it, and we can consider it for future updates. Thanks for your interest! 🚀

**Comment by SaadAshraf12** [comment]
I was thinking about using this app to make a system that detects the
drowsiness of a driver in car and sends an alarm through his/her android
phone.

**Comment by glenn-jocher** [comment]
@SaadAshraf12 That's a great application idea! While we don't provide the source code for the Ultralytics Android app, you can use the YOLOv5 models for your project. You would need to develop your own Android application that integrates the YOLOv5 model, possibly converted to TensorFlow Lite format for mobile deployment. For drowsiness detection, you would train a custom YOLOv5 model to recognize signs of driver fatigue. Once trained, you can deploy the model in your app to run real-time detection and trigger alarms. Good luck with your project, and if you have further questions on YOLOv5, feel free to ask! 😊👍
